File: android/DFM/src/main/java/com/dontflymoney/layout/MoveLine.java
package com.dontflymoney.layout;

import android.content.Context;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.dontflymoney.adapters.MoveAdapter;
import com.dontflymoney.baseactivity.SmartActivity;
import com.dontflymoney.view.R;

import org.json.JSONException;

import java.text.DateFormat;
import java.text.DecimalFormat;

public class MoveLine extends LinearLayout
{
	public MoveLine(Context context, AttributeSet attributeSet)
	{
		super(context, attributeSet);
	}

	@Override
	protected void onFinishInflate()
	{
		super.onFinishInflate();
		NameField = (TextView)findViewById(R.id.name);
		DateField = (TextView)findViewById(R.id.date);
		TotalField = (TextView)findViewById(R.id.value);
		CheckedField = (ImageView)findViewById(R.id.check_move);
	}

	private MoveAdapter.Move move;

	public TextView NameField;
	public TextView DateField;
	public TextView TotalField;
	public ImageView CheckedField;

	public void setMove(final SmartActivity activity, MoveAdapter.Move move, int color, boolean canCheck) throws JSONException
	{
		setBackgroundColor(color);

		this.move = move;

		NameField.setText(move.Description);
		setTotalField(move);
		setCheckField(move, canCheck);
		setDateField(move);

		setOnClickListener(new OnClickListener()
		{
			@Override
			public void onClick(View v)
			{
				activity.clickedView = v;
				v.showContextMenu();
			}
		});
	}

	private void setTotalField(MoveAdapter.Move move)
	{
		int totalColor = move.Total < 0 ? Color.RED : Color.BLUE;
		double totalToShow = move.Total < 0 ? -move.Total : move.Total;
		String totalStr = new DecimalFormat("#,##0.00").format(totalToShow);

		TotalField.setTextColor(totalColor);
		TotalField.setText(totalStr);
	}

	private void setCheckField(MoveAdapter.Move move, boolean canCheck)
	{
		if (canCheck)
		{
			int idResource = move.Checked ? R.drawable.green_sign : R.drawable.red_sign;
			CheckedField.setImageResource(idResource);
		}
		else
		{
			CheckedField.setVisibility(View.GONE);
		}
	}

	private void setDateField(MoveAdapter.Move move)
	{
		if (DateField != null)
		{
			DateFormat formatter = DateFormat.getDateInstance(DateFormat.SHORT);
			String dateInFull = formatter.format(move.Date.getTime());
			DateField.setText(dateInFull);
		}
	}



	public int getId()
	{
		return move.ID;
	}

	public String getDescription()
	{
		return move.Description;
	}

	public boolean isChecked()
	{
		return move.Checked;
	}



}

File: android/DFM/src/main/java/com/dontflymoney/layout/MoveLine.kt
package com.dontflymoney.layout

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.dontflymoney.adapters.MoveAdapter
import com.dontflymoney.view.ExtractActivity
import com.dontflymoney.view.R
import org.json.JSONException
import java.text.DateFormat
import java.text.DecimalFormat

class MoveLine(context: Context, attributeSet: AttributeSet) : LinearLayout(context, attributeSet) {

    override fun onFinishInflate() {
        super.onFinishInflate()
    }

    private var move: MoveAdapter.Move? = null

    val NameField: TextView get() = findViewById(R.id.name) as TextView
    val DateField: TextView get() = findViewById(R.id.date) as TextView
    val TotalField: TextView get() = findViewById(R.id.value) as TextView
    val CheckedField: ImageView get() = findViewById(R.id.check_move) as ImageView

    @Throws(JSONException::class)
    fun setMove(activity: ExtractActivity, move: MoveAdapter.Move, color: Int, canCheck: Boolean) {
        setBackgroundColor(color)

        this.move = move

        NameField.text = move.Description
        setTotalField(move)
        setCheckField(move, canCheck)
        setDateField(move)

        setOnClickListener { v ->
            activity.clickedView = v
            v.showContextMenu()
        }
    }

    private fun setTotalField(move: MoveAdapter.Move) {
        val totalColor = if (move.Total < 0) Color.RED else Color.BLUE
        val totalToShow = if (move.Total < 0) -move.Total else move.Total
        val totalStr = DecimalFormat("#,##0.00").format(totalToShow)

        TotalField.setTextColor(totalColor)
        TotalField.text = totalStr
    }

    private fun setCheckField(move: MoveAdapter.Move, canCheck: Boolean) {
        if (canCheck) {
            val idResource = if (move.Checked) R.drawable.green_sign else R.drawable.red_sign
            CheckedField.setImageResource(idResource)
        } else {
            CheckedField.visibility = View.GONE
        }
    }

    private fun setDateField(move: MoveAdapter.Move) {
        if (DateField != null) {
            val formatter = DateFormat.getDateInstance(DateFormat.SHORT)
            val dateInFull = formatter.format(move.Date.time)
            DateField!!.text = dateInFull
        }
    }


    override fun getId(): Int {
        return move!!.ID
    }

    val description: String
        get() = move!!.Description

    val isChecked: Boolean
        get() = move!!.Checked!!


}
